src/openrct2/world/Location.hpp

```diff
@@ -273,6 +273,11 @@ struct TileCoordsXYZ
         return !(*this == other);
     }
 
+    CoordsXYZ ToCoordsXYZ()
+    {
+        return { x * 32, y * 32, z * 8 };
+    }
+
```

Comment:
What I meant is that you had to create a method like the one on line 248 (with TileCoords and Coords reversed).

Reply:
But that will create a circular dependency between both

Reply:
Hm, that's where my knowledge of C++ hits its limits. @duncanspumpkin could you comment?

Reply:
In addition we were trying to avoid converting up and down between the types. That is why there wasn't anything like this before and why the constructor is explicit. I think this is fine but it should be discouraged from use.

Reply:
@duncanspumpkin Yes, but would it be possible to add an explicit constructor to create `Coords` from `TileCoords`?

Reply:
You wouldn't be able to resolve CoordsXY. So yes agree with @tupaschoal its not possible.